packages/cloud_firestore/cloud_firestore_platform_interface/lib/src/method_channel/method_channel_document_reference.dart

```diff
@@ -98,43 +98,36 @@ class MethodChannelDocumentReference extends DocumentReferencePlatform {
   Stream<DocumentSnapshotPlatform> snapshots({
     bool includeMetadataChanges = false,
   }) {
-    // It's fine to let the StreamController be garbage collected once all the
-    // subscribers have cancelled; this analyzer warning is safe to ignore.
-    late StreamController<DocumentSnapshotPlatform>
-        controller; // ignore: close_sinks
+    final observerId = Stream.fromFuture(
```

Comment:
probably should rename this to avoid confusion with the [function argument](https://github.com/FirebaseExtended/flutterfire/pull/8121/files#diff-3fde93986bda31f622bde969df19fd5ace5349d19f3f7cf3c4265b502f35f27dR107) of the same name.